### PR TITLE
Add Snowman/Iron Golem Spawn Eggs conditionally

### PR DESCRIPTION
**Pull request by glowredman**
Depends on and adds compat to https://github.com/GTNewHorizons/Hodgepodge/pull/398

### CODE REVIEW

File: src/main/java/codechicken/nei/api/ItemInfo.java [hunk 1]
```diff
@@ -458,8 +460,10 @@ else if (item == Items.fishing_rod || item == Items.flint_and_steel || item == I
     }
 
     private static void addSpawnEggs() {
-        addEntityEgg(EntitySnowman.class, 0xEEFFFF, 0xffa221);
-        addEntityEgg(EntityIronGolem.class, 0xC5C2C1, 0xffe1cc);
+        if (!Loader.isModLoaded("hodgepodge") || !TweaksConfig.addModEntityStats) {
```
Comment (mitchej123) [inline src/main/java/codechicken/nei/api/ItemInfo.java:463]: This isn't compatible with older versions of hodgepodge -- please add version identification or reflection to get this field

Comment (glowredman) [reply src/main/java/codechicken/nei/api/ItemInfo.java:463]: True, added. I set 2.5.44 as min version for now but I don't know in which HP version will include the PR. I need to change the version in `dependencies.gradle` anyway (currently uses -pre version), at that time I would change the version here too.

Comment (mitchej123) [reply src/main/java/codechicken/nei/api/ItemInfo.java:463]: Can you just only run this if hodgepodge of a specific version is present?  I don't want to add a requirement for a specific version of hodgepodge in NEI; past what lwjgl3ify/etc would force.  
`if(hodgepodge && hodgehodge.version > xxxx) {registerStuff)`

Mainly if you made the check for `addModEntityStats` use reflection, and default to false, this check here would work perfectly

**Review by mitchej123 — CHANGES_REQUESTED** [review]
Don't add a dep for a specific hodgepodge version, handle old ones gracefully

**Review by mitchej123 — APPROVED** [review]
Seems fine once the pre release dep is gone

**Comment by Dream-Master** [comment]
@mitchej123 can you approve the hodgepodge pr as well ?

**Comment by glowredman** [comment]
Replaced by https://github.com/GTNewHorizons/Hodgepodge/pull/398/commits/6923bb32965350873cc5b8c9e36fbe91f347aeba#diff-6d6ea392b5f8c6d98cbf5ca099fc56393c71b870222991be4b6b95edcb4a710fR102-R105

**Pull request closed without merging**